c2rust-analyze/src/main.rs

```diff
@@ -222,6 +382,68 @@ fn describe_span(tcx: TyCtxt, span: Span) -> String {
     format!("{}: {}{}{}", line, src1, src2, src3)
 }
 
+/// Return all `body_owners`, ordered according to a postorder traversal of the graph of references
+/// between bodies.
+fn walk_callgraph(tcx: TyCtxt) -> Vec<LocalDefId> {
+    let mut seen = HashSet::new();
+    let mut order = Vec::new();
+    enum Visit {
+        Pre(LocalDefId),
+        Post(LocalDefId),
+    }
+    let mut stack = Vec::new();
+
+    for root_ldid in tcx.hir().body_owners() {
+        if seen.contains(&root_ldid) {
+            continue;
+        }
+        stack.push(Visit::Pre(root_ldid));
+        while let Some(x) = stack.pop() {
+            match x {
```

Comment:
```suggestion
        while let Some(ldid) = stack.pop() {
            match ldid {
```

Reply:
`x` is a wrapper and not (just) a `LocalDefId`, so naming it `ldid` would be misleading IMO

Reply:
I think `ldid` makes more sense than `x`.  It always only contains an `ldid`, it's just either `Pre` or `Post`.

Reply:
This function takes a recursive algorithm (postorder graph traversal) and implements it iteratively.  `Visit` is best thought of not as "a `LocalDefId` plus an extra bit" (even though concretely that's exactly what it is), but as "a step of the original recursive algorithm".

Reply:
Can it be named `visit` then?

Reply:
OK, renamed to `visit`